include/cvc5/cvc5_proof_rule.h

```diff
@@ -2499,13 +2499,17 @@ enum ENUM(ProofRewriteRule)
   /**
    * \verbatim embed:rst:leading-asterisk
    * **Equality -- Macro lambda application eliminate shadow**
-   * 
+   *
    * .. math::
    *   ((\lambda x_1 \ldots x_n.\> t) \ t_1 \ldots t_n) = ((\lambda y_1 \ldots y_n.\> t \sigma) \ t_1 \ldots t_n)
-   * 
+   *
    * where :math:`\sigma` replaces :math:`x_1, \dots, x_n` by
    * :math:`y_1, \dots, y_n`. The terms may either be of kind
    * `cvc5::Kind::APPLY_UF` or `cvc5::Kind::HO_APPLY`.
+   * This rule ensures that :math:`y_1, \dots, y_n` are not in the free
+   * variables of :math:`t_1 \ldots t_n` and
+   * :math:`(\lambda x_1 \ldots x_n.\> t)` is alpha-equivalent to
```

Comment:
Isn't it the case that the lambda expressions are necessarily gonna be alpha-equivalent when `\sigma` is a valid renaming, which is the consequence of the condition above?

Reply:
Not if `t` had a variable in `y_1 ... y_n` that was not in `x_1 ... x_n`.

Reply:
Well, `t` will not, since `y1...yn` are not among the free variables of the *lambda expression*. If you properly constrain the substitution you'll gen an alpha equivalent term, which is what you want.

Reply:
It is more consice to say they are alpha equivalent, which implies this condition as well as saying that `y_1 ... y_n` are unique.

Reply:
You do not need to say that they are alpha equivalent if you say the renaming is such that it does not include free variables from the lambda expression. You could say:

```
This rule requires that :math:`y_a, \dots, y_n` are pairwise distinct and not in the free variables of :math:`(\lambda x_1 \ldots x_n.\> t)`.
````

Reply:
It's not that simple, I've made the description more accurate now.

Reply:
I see now. This rule is doing much more than I assumed it was. :) I like how you extended the documentation, thanks. I'd suggest to include an example, like the second one of the comment above. 

And maybe we should consider renaming the rule, because that's what misled me: I had assumed from the name that you were just eliminating shadowing among the variables bound by the lambda, but you are in fact renaming variables in the lambda expression to avoid capture of the variables in `t_1 ... t_n`.

Reply:
`MACRO_LAMBDA_CAPTURE_AVOID`?

Reply:
Sounds good!